### PR TITLE
Band pinned content enhancements

### PR DESCRIPTION
**Pull request by mikemai2awesome**
## Jira

http://vjira2:8080/browse/BDS-828

## Summary

Simplified pinned content feature in the band component for easier implementation..

## Details

- Grid component usage has been removed from the band component
- Band twig template is refactored to be simpler
- Deprecated grid props inside band
- Added new props to pass content into the band: `content` and `pinned_content`
- Fully revamped demo pages and documentation
- There will be breaking changes for the existing share tool implementation on the Drupal side (cc @charginghawk)

## How to test

Run the branch locally and check all the band demos and documentation.

### CODE REVIEW

**Comment by mikemai2awesome** [comment]
@remydenton Thanks! I will try the deprecation thing again and see if I run into errors.

I think we should create an adaptor, so let's sync up again once I finish on this.

File: packages/components/bolt-band/src/band.twig [hunk 1]
```diff
@@ -1,79 +1,148 @@
...
+{% set theme = theme in theme_options ? theme : schema.properties.theme.default %}
+{% set full_bleed = full_bleed is sameas(true) or full_bleed is sameas(false) ? full_bleed : schema.properties.full_bleed.default %}
 
-{% set size = size in schema.properties.size.enum ? size : schema.properties.size.default %}
-{% set size = size == "none" ? false : size %}
-{% set theme = theme in schema.properties.theme.enum ? theme : schema.properties.theme.default %}
-{% set theme = theme == "none" ? false : theme %}
+{# Deprecated #}
+{% if fullBleed %}
+  {% set fullBleed = full_bleed %}
```
[the remark this answers is not in the document]
Comment (remydenton) [reply packages/components/bolt-band/src/band.twig]: Also, we should move this section above where the validation happens.  For example, you'd want to do this

```twig
{% set full_bleed = fullBleed %}
```
_before_ you do this:

```twig
{% set full_bleed = full_bleed is sameas(true) or full_bleed is sameas(false) ? full_bleed : schema.properties.full_bleed.default %}
```

File: packages/components/bolt-band/src/band.twig [hunk 2]
```diff
@@ -1,79 +1,148 @@
...
-{% set componentName = "band" %}
-{% set baseClass = prefix ~ componentName %}
-{% set contentTag = contentTag in contentTags ? contentTag: "div" %}
-{% set fullBleed = fullBleed == false and fullBleed is not null ? false : true %}
+{# Deprecated #}
+{% if fullBleed %}
+  {% set full_bleed = fullBleed %}
+{% endif %}
+{% if contentTag %}
+  {% set contentTag = null %}
```
Comment (remydenton) [inline packages/components/bolt-band/src/band.twig]: I explicitly called out fullBleed, but the same is true for all the other deprecated variables.   For example, this should be 
```twig
{% set tag = contentTag %}
```
If you get to some that are too difficult to handle without a more complicated adaptor (possibly the rest), just put a comment that mentions which ones are not backwards compatible.  It don't see any value in setting them to `null`. We'll discuss next steps (how to release backward-breaking changes) in today's dev huddle.

**Comment by sghoweri** [comment]
Closing this one out in lieu of the 3 smaller PRs I broke this great work from @mikemai2awesome into:

1. Component-specific changes (confirming that the initial Band refactor work could be released via a minor vs major release of Bolt): #1088 
2. API and Demo-specific changes that depend on the work from #1088: #1091 
3. Misc Band-related cleanup that doesn't depend on #1088 or #1091: #1090

**Pull request closed without merging**